### ISSUE TITLE
Handle type clash

### ISSUE THREAD

**Issue by h4x3rotab** [paritytech/subxt#271]
https://polkadot.js.org/docs/api/start/types.extend#type-clashes

As mentioned in the above link from polkadotjs-api, there could be type clashes, and a well known example is `Balances::Balance = u128` and `Assets::Balance = u64`. In the current impl, subxt only handles `Balance = u128`, and will use this definition to decode `Assets::Balance`, causing a failure to decode the Assets events.

A simple approach like it in polkadotjs: in the type registry, let's add the pallet name for specific type look up. Usually the type definition don't require change, but for the type clashes, we can register a full type name `{pallet}::{type}` in the registry.

For example, we call:

```rust
event_type_registry.register_type_size::<u64>("Assets::Balance");
```

And in `events.rs`, whenever it looks up a type, try `{pallet}::{type}` before `{type}`.

**Comment by h4x3rotab** [comment]
Sample fix:  https://github.com/Phala-Network/phala-blockchain/pull/190/files

**Comment by en** [comment]
I faced with the same problem.
There might be a solution after this PR is merged.
https://github.com/paritytech/substrate/pull/8615

**Comment by ascjones** [comment]
Yes indeed, I'm back next week from holiday to work again on that and that should solve most custom type problems

**Comment by ascjones** [comment]
Closed by #294
